src/lib/badge/_badge-theme.scss

```diff
@@ -0,0 +1,177 @@
+@import '../core/theming/palette';
+@import '../core/theming/theming';
+@import '../core/typography/typography-utils';
+
+$mat-badge-font-size: 12px;
+$mat-badge-font-weight: 600;
+$mat-badge-size: 22px !default;
+
+.mat-badge {
+  position: relative;
+}
+
+.mat-badge:after,
+.mat-badge-icon,
+.mat-badge-text,
+.mat-badge-svg-icon {
+  position: absolute;
```

Comment:
All of these positioning and alignment styles should be in the `badge.scss`, not in the theme. The theme is only for color-related properties.

Reply:
I agree, however, since this is a directive there is no way to include styles w/ the component. Do you have a good solution for that?

Reply:
You can make it a component that has a template of `<ng-content>`. Having these styles in the theme will cause them to be loaded multiple times for pages with multiple themes.

Reply:
Ya, thats what I had. You can't do multiple components on the same element though, example: icon w/ a badge.

Reply:
You can't, but you can have an icon inside a badge.